### PR TITLE
Demo for tokens component

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-tokens-demo](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-tokens-demo)

File: src/app/docs/tokens/tokens-docs.component.html [hunk 1]
```diff
@@ -7,17 +7,52 @@
...
+        >
+        </sky-docs-demo-control-panel-checkbox>
+        <sky-docs-demo-control-panel-checkbox
+          label="Dismissible"
+          propertyName="dismissible"
+          [checked]="true"
+        >
+        </sky-docs-demo-control-panel-checkbox>
+        <sky-docs-demo-control-panel-checkbox
+          label="Focusable"
```
Comment (blackbaud-johnly) [inline src/app/docs/tokens/tokens-docs.component.html:28]: Is it correct behavior that the when both the focusable and dismissible options are selected and you tab through the tokens, the focus alternates between the tokens and the close button?

Also, should the *Reset* option reload tokens that have been closed?

Comment (Blackbaud-AlexKingman) [reply src/app/docs/tokens/tokens-docs.component.html:28]: Yes, that's by design.

Good idea about the reset button. I'll implement that.

File: src/app/public/plugin-resources/code-examples/tokens/basic/tokens-demo.component.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import {
+  Component
+} from '@angular/core';
+
+import {
+  SkyToken
+} from '@skyux/indicators';
+
+@Component({
+  selector: 'sky-tokens-demo',
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/tokens/basic/tokens-demo.component.ts]: Yep good catch. done.

File: src/app/public/plugin-resources/code-examples/tokens/custom/tokens-demo.component.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  SkyTokensMessage,
+  SkyTokensMessageType
+} from '@skyux/indicators';
+
+import {
+  Subject
+} from 'rxjs';
+
+@Component({
+  selector: 'sky-tokens-demo',
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/plugin-resources/code-examples/tokens/custom/tokens-demo.component.ts]: Done.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-tokens-demo](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-tokens-demo)

File: src/app/docs/tokens/tokens-docs.component.ts [hunk 4]
```diff
@@ -1,9 +1,62 @@
 import {
   Component
 } from '@angular/core';
+import { SkyDocsDemoControlPanelChange } from '@skyux/docs-tools';
```
Comment (Blackbaud-SteveBrush) [inline src/app/docs/tokens/tokens-docs.component.ts:4]: Import styling.

File: src/app/docs/tokens/tokens-docs.component.ts [hunk 5]
```diff
@@ -1,9 +1,62 @@
...
+    if (change.focusable !== undefined) {
+      this.demoSettings.focusable = change.focusable;
+    }
+  }
+
+  public onDemoReset(): void {
+    this.colors = this.parseTokens(this.defaultColors);
+  }
+
+  private parseTokens(data: any[]): SkyToken[] {
```
Comment (Blackbaud-SteveBrush) [inline src/app/docs/tokens/tokens-docs.component.ts:54]: This is my mistake, but "parse" only deals with converting `string` data into something else, so it doesn't make much sense to use here (I learned this recently, after adding that word to a lot of our docs-tools methods).

I'd change it to `getTokens` or `convertToTokens`.

File: src/app/public/plugin-resources/code-examples/tokens/basic/tokens-demo.component.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    { name: 'Turquoise' },
+    { name: 'White' },
+    { name: 'Yellow' }
+  ];
+
+  constructor() {
+    this.colors = this.parseTokens(this.defaultColors);
+  }
+
+  private parseTokens(data: any[]): SkyToken[] {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/plugin-resources/code-examples/tokens/basic/tokens-demo.component.ts:35]: Same comment as above RE: "parse".

File: src/app/public/plugin-resources/code-examples/tokens/custom/tokens-demo.component.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    if (!this.tokensController) {
+      this.tokensController = new Subject<SkyTokensMessage>();
+    }
+
+    this.tokensController.next({
+      type: SkyTokensMessageType.FocusLastToken
+    });
+  }
+
+  private parseTokens(data: any[]): SkyToken[] {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/plugin-resources/code-examples/tokens/custom/tokens-demo.component.ts:86]: Same comment as above, RE: "parse".

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-tokens-demo](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-tokens-demo)

**Review by blackbaud-johnly — APPROVED** [review]

**Pull request merged**